src/app/effects/courses.effects.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable } from '@angular/core';
+import { Actions, Effect, ofType } from '@ngrx/effects';
+import { Observable, of } from 'rxjs';
+import { GetUserFailure, GetUserSuccess, LoginActions, LoginActionTypes, LoginFailure, LoginSuccess } from '../actions/login.actions';
+import { AuthService } from '../services/auth.service';
+import { catchError, map, mergeMap } from 'rxjs/operators';
+import { Router } from '@angular/router';
+import {
+  AddCourseError,
+  AddCourseSuccess,
+  CoursesActionTypes, EditCourseError, EditCourseSuccess,
+  LoadCourses,
+  LoadCoursesError,
+  LoadCoursesSuccess,
+  RemoveCourseError,
+  RemoveCourseSuccess,
+} from '../actions/courses.actions';
+import { CoursesService } from '../services/courses.service';
+
+@Injectable()
+export class CoursesEffects {
+
+  @Effect()
+  loadCourses$: Observable<any> = this.actions$.pipe(
+    ofType(CoursesActionTypes.LoadCourses),
+    mergeMap(action =>
+      this.coursesService.getList(action['start'], action['count'], action['searchText']).pipe(
+        map((res) => {
+          return new LoadCoursesSuccess(res);
+        }),
+        catchError(err => of(new LoadCoursesError(err))),
+      ),
+    ),
+  );
```

Comment:
You can avoid having to access action object properties via brackets `action['start']` just by helping TypeScript compiler a bit to understand that this is `LoadCourses` object like this: `mergeMap((action: LoadCourses) => ...`. Also I don't know about you but to me, having second nested pipe operator makes the code harder to read and understand. You can get the same effect with just one pipe:

     loadCourses$: Observable<any> = this.actions$.pipe(
        ofType(CoursesActionTypes.LoadCourses),
        mergeMap((action: LoadCourses) =>
          this.coursesService.getList(action.start, action.count, action.searchText)),
        map((res) => {
          return new LoadCoursesSuccess(res);
        }),
        catchError(err => of(new LoadCoursesError(err))),
      );

Reply:
thxs  fixed